### PR TITLE
Add npm tasks with Artifactory

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: jfrog-utils/jutils.js [hunk 1]
```diff
@@ -104,6 +105,17 @@ function addBoolParam(cliCommand, inputParam, cliParam) {
     return cliCommand
 }
 
+// Determines the required working directory for running the cli.
+// Decision is based on the default path to run, and the provided path by the user.
+function determineCliWorkDir(defaultPath, providedPath) {
```
Comment (eyalbe4) [inline jfrog-utils/jutils.js]: The if statement in this function can be probably simplified to:
```
    if (providedPath) {
        if (path.isAbsolute(providedPath)) {
            return providedPath;
        }
        return path.join(defaultPath, providedPath);
    }
    return defaultPath;
```
Also, this function seems to be used by the NPM task only. It is therefore better to place it in *npmBuild.js*.

File: ArtifactoryNpm/task.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
+{
+    "id": "07fc5c20-6772-11d2-cace-6501d1a00335",
+    "name": "ArtifactoryNpm",
+    "friendlyName": "Artifactory npm",
+    "description": "Install, pack and deploy npm projects to Artifactory",
```
Comment (eyalbe4) [inline ArtifactoryNpm/task.json]: Install, pack and deploy npm **packages** to Artifactory

File: ArtifactoryNpm/task.json [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                "publish": "publish"
+            }
+        },
+        {
+            "name": "artifactoryService",
+            "type": "connectedService:artifactoryService",
+            "label": "Artifactory service",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Artifactory service used for npm tasks."
```
Comment (eyalbe4) [inline ArtifactoryNpm/task.json]: The Artifactory service used by the task.

File: ArtifactoryNpm/task.json [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            "required": false,
+            "helpMarkDown": "A list of npm arguments and options in the form of \"--arg1=value1 --arg2=value2\"."
+        },
+        {
+            "name": "collectBuildInfo",
+            "type": "boolean",
+            "label": "Collect build info",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Select to collect build info."
```
Comment (eyalbe4) [inline ArtifactoryNpm/task.json]: Let's change this *helpMarkDown* label in all
Select to collect build info and store it locally. The build info can be later published to Artifactory using the "Artifactory Publish Build Info" task.

File: ArtifactoryNpm/task.json [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            "label": "Command",
+            "defaultValue": "install",
+            "helpMarkDown": "The npm command to run.",
+            "required": "true",
+            "properties": {
+                "EditableOptions": "False"
+            },
+            "options": {
+                "install": "install",
+                "publish": "publish"
```
[the remark this answers is not in the document]
Comment (barbelity) [reply ArtifactoryNpm/task.json]: Done

File: jfrog-utils/jutils.js (hunk 1 above)
Comment (barbelity) [reply jfrog-utils/jutils.js]: Done

File: ArtifactoryNpm/task.json (hunk 2 above)
Comment (barbelity) [reply ArtifactoryNpm/task.json]: Done

File: ArtifactoryNpm/task.json (hunk 3 above)
Comment (barbelity) [reply ArtifactoryNpm/task.json]: Done

File: ArtifactoryNpm/task.json (hunk 4 above)
Comment (barbelity) [reply ArtifactoryNpm/task.json]: Done

**Pull request merged**